curdleproofs/curdleproofs/msm_accumulator.py

```diff
@@ -41,38 +32,37 @@ def compute_MSM(
 class MSMAccumulator:
     def __init__(self) -> None:
         self.A_c = Z1
-        self.base_scalar_map: Dict[Tuple[int, int], Fr] = {}
+        self.base_scalar_map: Dict[bytes, Scalar] = {}
 
     def accumulate_check(
         self,
-        C: PointProjective,
-        bases: List[PointProjective],
-        scalars: Union[List[Fr], List[int]],
+        C: G1Point,
+        bases: List[G1Point],
+        scalars: List[Scalar],
     ) -> None:
-        random_factor = Fr(random.randint(1, Fr.field_modulus))
+        random_factor = random_scalar()
 
-        self.A_c = add(self.A_c, multiply(C, int(random_factor)))
+        self.A_c = self.A_c + C * random_factor
 
         for (base, scalar) in zip(bases, scalars):
-            # print("base", base)
-            if is_inf(base):
+            # TODO: Is this infinity check necessary for G1Point class?
```

Comment:
I think it's an optimization because zero bases contribute nothing to the MSM. We can leave it here IMO.